modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarCollapse.es.js

```diff
@@ -0,0 +1,147 @@
+import ClayIcon from '../shared/ClayIcon.es';
+import CriteriaSidebarItem from './CriteriaSidebarItem.es';
+import getCN from 'classnames';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {jsDatetoYYYYMMDD} from '../../utils/utils.es';
+import {PROPERTY_TYPES} from '../../utils/constants.es';
+
+/**
+ * Returns a default value for a property provided.
+ * @param {Object} property
+ * @returns
+ */
+function getDefaultValue(property) {
+	const {options, type} = property;
+
+	let defaultValue = '';
+
+	if (type === PROPERTY_TYPES.STRING && options && options.length) {
+		defaultValue = options[0].value;
+	}
+	else if (type === PROPERTY_TYPES.DATE) {
+		defaultValue = jsDatetoYYYYMMDD((new Date()));
+	}
+	else if (type === PROPERTY_TYPES.DATE_TIME) {
+		defaultValue = (new Date()).toISOString();
+	}
+	else if (type === PROPERTY_TYPES.BOOLEAN) {
+		defaultValue = 'true';
+	}
+	else if (type === PROPERTY_TYPES.INTEGER) {
+		defaultValue = 0;
+	}
+	else if (type === PROPERTY_TYPES.DOUBLE) {
+		defaultValue = '0.00';
+	}
+
+	return defaultValue;
+}
+
+class CriteriaSidebarCollapse extends Component {
+	static propTypes = {
+		onCollapseClick: PropTypes.func,
+		propertyGroups: PropTypes.array,
+		propertyKey: PropTypes.string,
+		searchValue: PropTypes.string
+	};
+
+	_filterProperties = searchValue => {
+		const propertyGroup = this.props.propertyGroups.find(
+			propertyGroup => this.props.propertyKey === propertyGroup.propertyKey
+		);
+
+		const properties = (propertyGroup ? propertyGroup.properties : []);
+
+		return properties.filter(
+			property => {
+				const propertyLabel = property.label.toLowerCase();
+
+				return propertyLabel.includes(searchValue.toLowerCase());
+			}
+		);
+	}
+
+	_handleClick = (key, editing) => () => {
```

Comment:
No need to create this method, you can pass `onCollapseClick` straight through

Reply:
I think I need this method because `_handleClick` it's a high order function, that receives `key` and `editing` and returns a function